### PR TITLE
feat: put new /chains endpoint behind feature flag

### PR DESCRIPTION
**Pull request by pitasi**
This PR restores the original `/chains` endpoint we had before #118, since this requires a cockroachdb version that currently we don't have in staging.

We can now proceed with api-server deployments while we wait for cockroach to be upgraded in staging, and not remain stuck 😁 

I introduced a simple `fflag` package to manage feature flags, it's the simplest thing I could think of but I think it's sufficient for now. We can move into other libraries or even microservice if we want to do fancy things in the future.

### CODE REVIEW

File: api/chains/chains.go [hunk 1]
```diff
@@ -37,6 +38,10 @@ const (
 	devTeamBudget            = "budget-dev-team"
 )
 
+const (
+	NewChainsFlag = "newchains"
```
Comment (tbruyelle) [inline api/chains/chains.go]: I would rather have something more meaningful like `retroCompatStagingDB`, and by the way inverse its behavior. 

By default it's false, so it's the new behavior, and we enable it only for staging. By doing this we avoid forgetting to activate it on other env. WDYT?

File: lib/fflag/fflag.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+// Enabled returns true if a particular feature flag is enabled.
+func Enabled(c *gin.Context, name string) bool {
+	if v, found := c.GetQuery(name); found {
+		return strings.ToLower(v) == "true"
+	}
+
+	return globalFlags[name]
+}
```
Comment (tbruyelle) [inline lib/fflag/fflag.go:35]: Nice tiny package, too bad it has a gin dependency :D

Comment (pitasi) [reply lib/fflag/fflag.go:35]: 😅

File: api/chains/chains.go (hunk 1 above)
Comment (pitasi) [reply api/chains/chains.go]: yeah we can do this way, I don't have any opinion on that :) I changed it [here](https://github.com/EmerisHQ/demeris-api-server/pull/128/commits/56e8f99552029e190216aba3c0a5e66854f0e1da)

Comment (tbruyelle) [reply api/chains/chains.go]: Well, for such thing I tend to make the default behavior according to the majority. Too many times it has hurt me when I forgot to set this kind of flag after a urgent production deployment :)

Thanks for the fix

**Review by tbruyelle — APPROVED** [review]

**Review by DeshErBojhaa — APPROVED** [review]

**Pull request merged**